code/clue.js

```diff
@@ -41,6 +160,12 @@ const randomSelector = array => {
 // With a killer, a weapon and a room.
 // The values will be set later.
 
+let mystery = {
+ suspects, 
+ weapons, 
+ rooms: []
+}
```

Comment:
This mystery object is indeed mysterious! I'm not completely sure but it looks like it now is an object ("mystery") containing the complete arrays with suspects, weapons and rooms - which is not what we need here. The mystery object should just contain one killer, weapon and room looking at the requirements. In the pickKiller, pickWeapon and pickRoom functions we randomly pick and assign that random pick to the mystery object, and later on the mystery object is what we reveal. No need to change any code in those functions!

This is how the mystery object code could look (with line breaks, I cannot get those to work..): 
`const mystery = {
  killer: null,
  weapon: null,
  room: null
}`

We can use const as the object keys will not change, even though the values will. Null means we set the initial value to no value I think.

Reply:
Great comment Annika! I agree 100%